include/alpaka/exec/ExecGpuCudaRt.hpp

```diff
@@ -87,6 +87,14 @@ namespace alpaka
 #if BOOST_ARCH_PTX && (BOOST_ARCH_PTX < BOOST_VERSION_NUMBER(2, 0, 0))
     #error "Cuda device capability >= 2.0 is required!"
 #endif
+
+// with clang it is not possible to query std::result_of for a pure device lambda created on the host side
+#ifndef BOOST_COMP_CLANG_CUDA BOOST_COMP_CLANG
```

Comment:
This should most probably be `#if BOOST_COMP_CLANG_CUDA && BOOST_COMP_CLANG`

Reply:
thx for the hint, I updated the PR